File: Python/Raw/oop-python/studentinfo.py
class Student:
    
    __totalCGPA = 0.0
    __totalCredits = 0

    def __init__(self, id, name, address):
        self.__studentId = id
        self.__studentName = name 
        self.__studentAddress = address 

    def getStudentId(self):
        return self.__studentId

    def getStudentName(self):
        return self.__studentName

    def getStudentAddress(self):
        return self.__studentAddress

    def getCgpa(self):
        cgpa =  self.__totalCGPA / self.__totalCredits
        if(self.__totalCredits == 0.0):
            return 0.0
        return cgpa

    def setStudentAddress(self, address):
        self.__studentAddress = address

    def getTotalCredits(self):
        return self.__totalCredits

    def addGrade(self, grade, credits):
        numericGrade = 0.0
        if (grade == "A+"):
            numericGrade = 4.0
        elif (grade == "A"):
            numericGrade = 3.75
        elif (grade == "A-"):
            numericGrade = 3.50
        elif (grade == "B+"):
            numericGrade = 3.25
        elif (grade == "B"):
            numericGrade = 3.00
        elif (grade == "B-"):
            numericGrade = 2.75
        elif (grade == "C+"):
            numericGrade = 2.50
        elif (grade == "C"):
            numericGrade = 2.25
        elif (grade == "D"):
            numericGrade = 2.00
        elif (grade == "F"):
            numericGrade = 0.00
        else:
            return None
        self.__totalCredits += credits
        self.__totalCGPA += (numericGrade * credits)

    #In java this one called toString
    def __str__(self):
     return str(self.__studentId) + " " +self.__studentName + " " + self.__studentAddress + " " +str(self.getCgpa()) + " " + str(self.getTotalCredits())



s = Student(100, 'Manik Mia', "Banani")
s.addGrade("A+", 3.00)
s.addGrade("A", 1.00)
s.addGrade("B+", 3.00)
s.addGrade("B", 3.00)
s.setStudentAddress("Nikunja")
print(s)
# print(s.getStudentName(), s.getCgpa())
# For 72 number line we use returning a string representation __str__
# print(s.__str__())

File: JavaScript/Raw/oop-js/studentinfo.js
class Student{
    #studentId = 0;
    #studentName = "";
    #studentAddress= "";
    #totalCGPA = 0.0;
    #totalCredits = 0;

    constructor(id, name, address){
        this.#studentId = id;
        this.#studentName = name;
        this.#studentAddress = address;
    }

    getStudentId(){
        return this.#studentId;
    }

    getStudentName(){
        return this.#studentName;
    }

    getStudentAddress(){
        return this.#studentAddress;
    }

    getCgpa(){
        var cgpa =  this.#totalCGPA / this.#totalCredits;
        if(this.#totalCredits == 0.0){
            return 0.0;
        }
        return cgpa;
    }

    setStudentAddress(address){
        this.#studentAddress = address;
    }

    getTotalCredits(){
        return this.#totalCredits
    }

    addGrade(grade, credits){
        var numericGrade = 0.0;
        switch (grade) {
            case "A+": numericGrade = 4.00; break;
            case "A" : numericGrade = 3.75; break;
            case "A-": numericGrade = 3.50; break;
            case "B+": numericGrade = 3.25; break;
            case "B" : numericGrade = 3.00; break;
            case "B-": numericGrade = 2.75; break;
            case "C+": numericGrade = 2.50; break;
            case "C" : numericGrade = 2.25; break;
            case "D" : numericGrade = 2.00; break;
            case "F" : numericGrade = 0.00; break;
        }
        this.#totalCredits += credits;
        this.#totalCGPA += (numericGrade * credits);
    }

    toString() {
        return this.#studentId + " " + this.#studentName + " " + this.#studentAddress + " " + this.getCgpa();
    }
}

s = new Student(100, 'Manik Mia', "Banani");
s.addGrade("A+", 3.00);
s.addGrade("A", 1.00);
s.addGrade("B+", 3.00);
s.addGrade("B", 3.00);
s.setStudentAddress("Nikunja");
console.log(s+" ");
// console.log(s.getStudentId(), s.getStudentName(), s.getCgpa(), s.getStudentAddress(), s.getTotalCredits());
// For 72 line we use toString
